### PR TITLE
Jruby fixes

### PR DESCRIPTION
**Pull request by pnomolos**
A few initial fixes for jruby support (Sequel spec was failing currently).  This should also fix https://github.com/seamusabshere/upsert/issues/61

### CODE REVIEW

**Comment by pnomolos** [comment]
This will have to wait on #79 for the mysql tests to pass (or I can pull that commit in).

File: lib/upsert/connection/jdbc.rb [hunk 1]
```diff
@@ -34,6 +35,10 @@ def execute(sql, params = nil)
           setters = self.class.const_get(:SETTER)
           statement = metal.prepareStatement sql
           params.each_with_index do |v, i|
+            if v.is_a?(Fixnum) && v > 2_147_483_647
+              next statement.setLong i+1, v
```
Comment (seamusabshere) [inline lib/upsert/connection/jdbc.rb]: break this into 2 lines?

```
foo
next
```

(cause we don't use the return val)

**Comment by pnomolos** [comment]
@seamusabshere I tried to get `rbx` working as well but looks like it's broken (and not really fixable).  I'm hoping travis fixes things on their end soon so we can have it be useful in the build matrix.

File: lib/upsert/connection/Java_ComMysqlJdbc_JDBC4Connection.rb [hunk 2]
```diff
@@ -6,9 +6,12 @@ class Connection
     class Java_ComMysqlJdbc_JDBC4Connection < Connection
       include Jdbc
 
-      # ? backtick?
       def quote_ident(k)
-        DOUBLE_QUOTE + k.to_s.gsub(DOUBLE_QUOTE, '""') + DOUBLE_QUOTE
+        if metal.useAnsiQuotedIdentifiers
+          DOUBLE_QUOTE + k.to_s.gsub(DOUBLE_QUOTE, '""') + DOUBLE_QUOTE
+        else
+          BACKTICK + k + BACKTICK
```
Comment (seamusabshere) [inline lib/upsert/connection/Java_ComMysqlJdbc_JDBC4Connection.rb]: do we need to `gsub` backticks? (probably a crazy edge case)

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -44,7 +44,7 @@ def new_connection
     ActiveRecord::Base.establish_connection :adapter => 'postgresql', :database => DATABASE, :username => CURRENT_USER
 
   when 'mysql'
-    password_argument = (PASSWORD.empty?) ? "" : "-p#{PASSWORD}"
+    password_argument = (PASSWORD.nil?) ? "" : "--password='#{PASSWORD.gsub(/'/, "'\''")}'"
```
Comment (seamusabshere) [inline spec/spec_helper.rb]:
```ruby
require 'shellwords'
Shellwords.escape 'xyz'
```

File: upsert.gemspec [hunk 4]
```diff
@@ -32,7 +32,7 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency 'rake', '~>10.1.1'
 
   if RUBY_VERSION >= '1.9'
-    gem.add_development_dependency 'activerecord-import'
+    gem.add_development_dependency 'activerecord-import', '0.3.1'
```
Comment (seamusabshere) [inline upsert.gemspec]: add a comment why we're pinning to this version?

File: lib/upsert/connection/Java_ComMysqlJdbc_JDBC4Connection.rb (hunk 2 above)
Comment (pnomolos) [inline lib/upsert/connection/Java_ComMysqlJdbc_JDBC4Connection.rb]: Apparently you're allowed to (https://github.com/tgriesser/knex/issues/753).  Looks like you escape it by doubling-up the backtick.  I'll add that.

File: spec/spec_helper.rb (hunk 3 above)
Comment (pnomolos) [inline spec/spec_helper.rb]: Good call.

File: upsert.gemspec (hunk 4 above)
Comment (pnomolos) [inline upsert.gemspec]: I'll have to review why I did that.  Something with the import not running properly, I believe.  That might have been the version that was used implicitly prior and I just pinned it to make sure it wasn't updated.

Comment (pnomolos) [inline upsert.gemspec]: I included a note, but I went through the versions until I found the most recent that worked (0.11.0).  Something changed between it and 0.12.0 that is causing it to fail.  Also, Postgres support for UPSERT was added in 0.12.0 so the benchmarks may be much closer in that case now.

**Pull request merged**